utils/oc.py

```diff
@@ -213,18 +214,64 @@ def recycle_pods(self, dry_run, namespace, dep_kind, dep_resource):
         pods = self.get(namespace, 'Pods')['items']
 
         if dep_kind == 'Secret':
-            pods_to_recycle = [pod['metadata']['name'] for pod in pods
+            pods_to_recycle = [pod for pod in pods
                                if self.secret_used_in_pod(dep_name, pod)]
         else:
             raise RecyclePodsUnsupportedKindError(dep_kind)
 
+        recyclables = {}
         for pod in pods_to_recycle:
-            logging.info(['recycle_pod', namespace, pod])
-            if not dry_run:
-                self.delete(namespace, 'Pod', pod)
-                logging.info(['validating_pods', namespace])
-                self.validate_pods_ready(
-                    namespace, self.secret_used_in_pod, dep_name)
+            if pod['status']['phase'] == 'Succeeded':
```

Comment:
That's currently handled by the definition of supported_recyclables.  Since Jobs and CronJobs are not listed as kinds that are recyclable they won't be handled at all.  I suppose we could special case Jobs and CronJobs to not even be in the map.  I can't say I feel strongly either way.

Reply:
the only pods i'm aware of that "finish" running are ones that belong to jobs.

so what i'm trying to ask is - is this really required if jobs are out of the recyclables list?

Reply:
I redid the logic here and removed that check.  You're right, it isn't needed to check if the master isn't one of the recyclables.